### ISSUE TITLE
`VMService::dispose` will complete outstanding requests with `RPCError`s that lack stack traces

### ISSUE THREAD

**Issue by andrewkolos** [dart-lang/sdk#56491]
Discovered while investigating a crasher of the flutter CLI tool: https://github.com/flutter/flutter/issues/153471.

https://github.com/dart-lang/sdk/commit/a7d8707a5964e174f2ce321d075e96e37a9a16ad modified `VMService` to complete any outstanding requests with errors upon the VM service connection closing. This was achieved by calling `VMService::dispose` upon the connection closing, which in turn will complete any outstanding request with an `RPCError`:

https://github.com/dart-lang/sdk/blob/6c92babf4d3a256c15d98f126a5a6c30ed148c7a/pkg/vm_service/lib/src/vm_service.dart#L1739-L1745

Notice that no stack trace is provided. Judging by https://github.com/dart-lang/sdk/commit/9a755f22278cf1d345dd758ce7c162d9b820a51c, should the above code instead be

```dart
      request.completeError(RPCError(
        request.method,
        RPCErrorKind.kServerError.code,
        'Service connection disposed',
      ));
```
?

cc @bkonyi

**Comment by bkonyi** [comment]
@derekxu16, can you take a look at this today and prepare a `package:vm_service` release with the fix?

**Comment by derekxu16** [comment]
Thanks for the detailed report @andrewkolos! I've just published `package:vm_service 14.2.5` which contains the fix: https://pub.dev/packages/vm_service/versions/14.2.5.
